### PR TITLE
bpo-35134: Move non-limited C API files to Include/cpython/

### PR DESCRIPTION
**Pull request by nw0**
Move
Include/{odictobject.h,parser_interface.h,picklebufobject.h,pydebug.h,pyfpe.h}
into Include/cpython/.

In the same vein as previous commits by @vstinner.

https://bugs.python.org/issue35134

### CODE REVIEW

File: Misc/NEWS.d/next/C API/2021-02-18-18-46-42.bpo-35134.dFpEDT.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Move odictobject.h, parser_interface.h, picklebufobject.h, pydebug.h, and
+pyfpe.h into the cpython/ directory. They must not be included directly, as
+they are already included by Python.h: :ref:`Include Files <api-includes>`.
```
Comment (vstinner) [inline Misc/NEWS.d/next/C API/2021-02-18-18-46-42.bpo-35134.dFpEDT.rst:3]: Hum. If we move more header files, the risk of breaking 3rd C extensions is higher. I'm still supportive of this change, but please document it also at C API > Porting to Python 3.10:
https://docs.python.org/dev/whatsnew/3.10.html#id2

In What's New in Python 3.10, document also the other header files that you already moved.

File: Parser/peg_api.c [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#include "parser_interface.h"
+#include "cpython/parser_interface.h"
```
Comment (vstinner) [inline Parser/peg_api.c]: Can't we include Python.h here instead?

File: Makefile.pre.in [hunk 3]
```diff
@@ -1051,17 +1051,13 @@ PYTHON_HEADERS= \
...
 		$(srcdir)/Include/opcode.h \
 		$(srcdir)/Include/osdefs.h \
 		$(srcdir)/Include/osmodule.h \
 		$(srcdir)/Include/patchlevel.h \
-		$(srcdir)/Include/picklebufobject.h \
 		$(srcdir)/Include/pycapsule.h \
-		$(srcdir)/Include/pydebug.h \
 		$(srcdir)/Include/pydtrace.h \
 		$(srcdir)/Include/pyerrors.h \
-		$(srcdir)/Include/pyfpe.h \
```
Comment (vstinner) [inline Makefile.pre.in:1064]: FYI I created https://bugs.python.org/issue43250 about pyfpe.h (I would like to remove it, but I suggest to first merge your PR ;-)).

File: Misc/NEWS.d/next/C API/2021-02-18-18-46-42.bpo-35134.dFpEDT.rst (hunk 1 above)
Comment (nw0) [reply Misc/NEWS.d/next/C API/2021-02-18-18-46-42.bpo-35134.dFpEDT.rst:3]: Updated. I assume the main concern is if they have been included directly? I've worded my note with that in mind, but let me know if I should add anything else.

I've taken a quick look at the remaining files: there are still more than 10 header files that could be moved into cpython/ wholesale, and about the same number which require some surgery. Would you support making these changes (hopefully to close bpo-35134 entirely) within 3.10?

**Comment by vstinner** [comment]
Merged, thanks.

**Pull request merged**